airbyte-integrations/connectors/source-google-ad-manager/source_google_ad_manager/source.py

```diff
@@ -0,0 +1,51 @@
+from datetime import datetime
+from typing import Any, Mapping, Tuple, List
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from .authenticator import GoogleAdManagerAuthenticator
+from .streams import AdUnitPerHourReportStream, AdUnitPerReferrerReportStream
+from .utils import convert_time_to_dict
+
+APPLICATION_NAME = 'spiny'
+
+
+class SourceGoogleAdManager(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        this check if the connector can connect to the google ad manager api with the credentials provided by the user
+        """
+        try:
+            google_ad_manager_authenticator = self._get_authenticator(config=config)
+            networks = google_ad_manager_authenticator.get_networks()
+            for network in networks:
+                logger.info('Network with network code "{}" and display name "{}" was found. at the following timezone "{}"'.format(network['networkCode'], network['displayName'], network["timeZone"]))
+            return True, None
+        except Exception as e:
+            return False, e
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        TODO: Replace the streams below with your own streams.
+
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        google_ad_manager_authenticator = self._get_authenticator(config=config)
+        google_ad_manager_client = google_ad_manager_authenticator.get_client()
+        today_date = datetime.today()
+        start_date = convert_time_to_dict(today_date)
+        end_date = convert_time_to_dict(today_date)
```

Comment:
The connector will retrieve data only for one day?

Reply:
Please move this to config parameters.

Reply:
Yes, I am retrieving data for a day, but I can make the range dynamic for my use case.